routes/agents.py

```diff
@@ -0,0 +1,152 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from typing import List
+from .. import models, schemas
+from ..database import get_db  
+
+router = APIRouter(prefix="/api/agents", tags=["Agents"])
+
+#list all
+@router.get("/", response_model=List[schemas.AgentOut])
+def get_agents(db: Session = Depends(get_db)):
+    agents = (
+        db.query(
+            models.PhoneAgent.id,
+            models.PhoneAgent.name,
+            models.PhoneAgent.calls_handled,
+            models.PhoneAgent.status,
+            models.AppointmentSetter.appointments,
+        )
+        .join(models.AppointmentSetter, models.AppointmentSetter.agent_id == models.PhoneAgent.id)
+        .all()
+    )
+    return [schemas.AgentOut(
+        id=a.id,
+        name=a.name,
+        appointments=a.appointments,
+        calls_handled=a.calls_handled,
+        status=a.status
+    ) for a in agents]
+
+
+#get single agent
+@router.get("/{agent_id}", response_model=schemas.AgentOut)
+def get_agent(agent_id: int, db: Session = Depends(get_db)):
+    agent = (
+        db.query(
+            models.PhoneAgent.id,
+            models.PhoneAgent.name,
+            models.PhoneAgent.calls_handled,
+            models.PhoneAgent.status,
+            models.AppointmentSetter.appointments,
+        )
+        .join(models.AppointmentSetter, models.AppointmentSetter.agent_id == models.PhoneAgent.id)
+        .filter(models.PhoneAgent.id == agent_id)
+        .first()
```

Comment:
both the api are returning same data si why create a api for geting value of a single lemnet if you are already passing it in get all api